server/src/main/java/io/unitycatalog/server/persist/utils/PagedEntityLister.java

```diff
@@ -0,0 +1,90 @@
+package io.unitycatalog.server.persist.utils;
+
+import io.unitycatalog.server.exception.BaseException;
+import io.unitycatalog.server.exception.ErrorCode;
+import io.unitycatalog.server.persist.dao.*;
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.Predicate;
+import jakarta.persistence.criteria.Root;
+import org.hibernate.Session;
+import org.hibernate.query.Query;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+public class PagedEntityLister<T extends IdentifiableDAO> {
+
+    private final Class<T> entityClass;
+
+    public PagedEntityLister(Class<T> entityClass) {
+        this.entityClass = entityClass;
+    }
+
+    public static final Integer DEFAULT_PAGE_SIZE = 100;
+
+    public static Integer getPageSize(Optional<Integer> maxResults) {
+        return maxResults
+                .filter(x -> (x > 0))
+                .map(x -> Math.min(x, DEFAULT_PAGE_SIZE))
+                .orElse(DEFAULT_PAGE_SIZE);
+    }
+
+    public String getNextPageToken(List<T> entities, Optional<Integer> maxResults) {
+        if (entities == null || entities.isEmpty() || entities.size() < getPageSize(maxResults)) {
+            return null;
+        }
+        // return the last entity name
+        return entities.get(entities.size() - 1).getName();
+    }
+
+    public String getParentEntity() {
```

Comment:
it's better to move this method to `IdentifiableDAO`, call it `getParentIdColumnName` and make it return an optional